go113_test.go

```diff
@@ -3,14 +3,154 @@
 package errors
```

Comment:
This will need the `go1.13` build tag, since we will not be exporting these functions from `xerrors` for <go1.13.

Also, testing the functions in our own code is kind of silly anyways, because they are just a wrapper for the standard library functions. So, we’re actually testing code that is strictly not our domain.

Reply:
actually [go113_test.go](https://github.com/Sherlock-Holo/errors/blob/support-std-errors/go113_test.go#L1) has build tag

Reply:
:woman_facepalming:

Reply:
Yes, maybe the test around the `Is` and `As` only must check that the compatibility is in there.

I don't know if the best solution, but I think in something like that:

```go
func TestIs(t *testing.T) {
	sentinelError := stdlib_errors.New("sentinel error")
	wrap := Wrap(sentinelError, "wrap error")
	if !Is(wrap, sentinelError) {
		t.Errorf("Expected that '%v' error and the '%v' error should be of the same type", sentinelError, wrap)
	}
}

func TestAs(t *testing.T) {
	type myError struct {
		error
	}
	err := &myError{stdlib_errors.New("error")}
	wrap := Wrap(err, "wrap error")

	var tt *myError
	if !As(wrap, &tt) {
		t.Errorf("Expected that '%v' error and the '%v' error should be of the same type", err, wrap)
	}
}
```

Reply:
`TestIs` and `TestAs` not only check the compatibility, it also make sure `withStack` and `withMessage` can work with them.

Reply:
By contract, `withStack` and `withMessage` work with `Is` and `As` by implementing `interface{ Unwrap(error) error }`. This is something that is not within our scope to test. We test that they implement `Unwrap` according to contract, everything else is on `Is` and `As` to test.